ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx

```diff
@@ -0,0 +1,233 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+
+import NetworkGraph from './NetworkGraph';
+import {
+    CustomEdgeModel,
+    CustomModel,
+    CustomNodeModel,
+    DeploymentData,
+    ExtraneousNodeModel,
+    NetworkPolicyState,
+} from './types/topology.type';
+import { EdgeState } from './components/EdgeStateSelect';
+import { DisplayOption } from './components/DisplayOptionsSelect';
+import { Simulation } from './utils/getSimulation';
+import { getNodeById } from './utils/networkGraphUtils';
+import { createExtraneousNodes, createExtraneousEdges, graphModel } from './utils/modelUtils';
+
+export type Models = {
+    activeModel: CustomModel;
+    extraneousModel: CustomModel;
+};
+
+function getFilteredEdges(
+    edges: CustomEdgeModel[],
+    selectedNode: CustomNodeModel
+): CustomEdgeModel[] {
+    const filteredEdges: CustomEdgeModel[] = [];
+    if (selectedNode.data && selectedNode.data.type !== 'EXTRANEOUS') {
+        edges.forEach((edge) => {
+            const { source, target } = edge;
+            const { type } = selectedNode.data;
+            if (type === 'NAMESPACE' || type === 'EXTERNAL_GROUP') {
+                // if a namespace is selected, add children's node edges
+                const { children } = selectedNode;
+                if (children?.includes(source) || children?.includes(target)) {
+                    filteredEdges.push({ ...edge, visible: true });
+                }
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
```

Comment:
not sure why ts is failing here since i check against `EXTRANEOUS` above, so i just commented ts-ignore for now